py2k/record.py

```diff
@@ -27,35 +27,23 @@
                         update_optional_schema)
 
 
-class IterableAdapter:
-    def __init__(self, iterator_factory):
-        self.iterator_factory = iterator_factory
-
-    def __iter__(self):
-        return self.iterator_factory()
-
-
 class KafkaRecord(BaseModel):
     __key_fields__: set = {}
     __include_key__: bool = False
 
     @classmethod
     def from_pandas(cls, df: pd.DataFrame) -> List['KafkaRecord']:
-        records = df.to_dict('records')
+        return list(cls.iter_from_pandas(df))
 
-        if records:
-            return [cls(**item) for item in records]
+    @classmethod
+    def iter_from_pandas(cls, df: pd.DataFrame) -> Iterator['KafkaRecord']:
+        records = df.to_dict('records')
 
-        warnings.warn(
-            "Unable to create kafka model from an empty dataframe.")
-        return []
+        if not records:
+            warnings.warn(
+                "Unable to create kafka model from an empty dataframe.")
```

Comment:
Should we do something other than warning, e.g. return empty?

Reply:
We are doing this :-) This function returns an empty iterator. 

Here is the [test](https://github.com/AbsaOSS/py2k/pull/67/files#diff-0adb667cd397bd56edce10eec11e1b10821740a10d72bd148b09645fc9108968R172) to confirm :-)